text/0003-templates.md

```diff
@@ -0,0 +1,393 @@
+
+### Authoring
+
+The below shows an example of a simple `authoring` template.
+
+```json5
+{
+  "name": "@example/authoring-template",
+  "version": "0.1.0",
+  "pnpm": {
+    "template": "authoring"
+  },
+  "author": "Example Organization <team@organization.example>",
+  "license": "MIT",
+  "repository": "git@organization.example/example/project"
+}
+```
+
+A package referencing the `@example/authoring-template` above will have the following on-disk and in-memory representations.
+
+**On-Disk**
+
+```json5
+{
+  "name": "@example/react-components",
+  "pnpm": {
+    "templates": {
+      "authoring": ["@example/authoring-template@0.1.0"],
+    }
+  }
+}
+```
+
+**In-Memory and Publish Time**
+
+```json5
+{
+  "name": "@example/react-components",
+  "author": "Example Organization <team@organization.example>",
+  "license": "MIT",
+  "repository": "git@organization.example/example/project"
+}
+```
+
+### Toolkit
+
+The `pnpm.templates.toolkit` flavor copies the `dependencies` block of the template and spreads it into the `devDependencies` of the referencing package.
+
+Given the following template:
+
+```json5
+{
+  "name": "@example/react-toolkit",
+  "dependencies": {
+    "jest": "^29.4.3",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "redux": "^4.2.1"
+  }
+}
+```
+
+A package referencing the `@example/react-toolkit` above will have the following on-disk and in-memory representations.
+
+**On-Disk**
+
+```json5
+{
+  "name": "@example/react-components",
+  "pnpm": {
+    "templates": {
+      "toolkit": ["@example/react-toolkit@0.1.0"],
+    }
+  }
+}
+```
+
+**In-Memory and Publish Time**
+
+```json5
+{
+  "name": "@example/react-components",
+  "dependencies": {
+    "redux": "^4.2.1"
+  },
+  "devDependencies": {
+    "jest": "^29.4.3",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "redux": "^4.2.1"
+  }
+}
+```
+
+This allows package authors to share a reusable set of dependencies that can be imported or required in the referencing package. Dependencies of dependencies are not normally visible in this manner since pnpm sets up a semi-strict `node_modules` structure by default.
+
+### Catalog
+
+The `pnpm.templates.catalog` flavor allows packages to declare a dependency using a version specifier from the referenced template. The `dependencies` block of the template will be available to reference through the `catalog:` version specifier protocol.
+
+```json5
+{
+  "name": "@example/frontend-catalog",
+  "pnpm": {
+    "template": "catalog"
+  },
+  "dependencies": {
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "redux": "^4.2.0",
+    "react-redux": "^8.0.0"
+  }
+}
+```
+
+A package referencing the `@example/frontend-catalog` above will have the following on-disk and in-memory representations.
+
+**On-Disk**
+
+```json5
+{
+  "name": "@example/react-components",
+  "pnpm": {
+    "templates": {
+      "catalog": {
+        "example-frontend": "@example/frontend-catalog@0.1.0"
```

Comment:
do we really need an alias? wouldn't it be enough to reference by package name? also, in most cases I think just specifying `catalog:` should be enough.

Reply:
We could remove the alias.

At the moment there's 2 ways to refer to something.

1. The standard `package.json` dependency specifier syntax: Ex `^1.0.0`.
2. The template specifier syntax:

https://github.com/pnpm/rfcs/blob/714d1d81d86446412d3dc9a3fdea4144b4d080bb/text/0003-templates.md?plain=1#L281-L287

Removing the alias and allowing:

```json
{
  "pnpm": {
    "templates": {
      "catalog": {
        "@example/frontend-catalog": "0.1.0"
      }
    }
  }
}
```

Mean introducing a 3rd syntax just for catalogs. For example, we wouldn't allow `^0.1.0` for the example above.  I'm okay with that if we think the ergonomics of not having an alias is worthwhile.